src/otx/algo/visual_prompting/zero_shot_segment_anything.py

```diff
@@ -926,15 +926,19 @@ def load_reference_info(
             log.info(f"reference info saved at {path_to_directly_load} was successfully loaded.")
 
         else:
-            _infer_reference_info_root: Path = (
-                self.infer_reference_info_root
-                if self.infer_reference_info_root == self.infer_reference_info_root.absolute()
-                else Path(default_root_dir) / self.infer_reference_info_root
-            )
+            if str(self.infer_reference_info_root) == "../.latest/train":
```

Comment:
I am sorry that I don't have any good idea, but it looks like risky to have "../.latest/train".. Could you consider any better idea?

Reply:
It is following a default workspace policy, if running otx train/test without any update on `work_dir`, a structure in workspace is like below:
```shell
otx-workspace/zsl
├── .latest
│   ├── export -> ../20240704_060236
│   └── train -> ../20240704_060002
│       ├── best_checkpoint.ckpt -> otx-workspace/zsl/20240704_060002/checkpoints/epoch_000.ckpt
│       ├── checkpoints
│       │   └── epoch_000.ckpt
│       ├── configs.yaml
│       ├── csv
│       │   └── version_0
│       │       ├── hparams.yaml
│       │       └── metrics.csv
│       ├── last.ckpt
│       ├── reference_infos
│       │   ├── reference_info.pickle
│       │   └── reference_info.pt
│       └── tensorboard
├── 20240704_060002 # train
│   ├── best_checkpoint.ckpt -> otx-workspace/zsl/20240704_060002/checkpoints/epoch_000.ckpt
│   ├── checkpoints
│   ├── configs.yaml
│   ├── csv
│   ├── last.ckpt
│   ├── reference_infos
│   └── tensorboard
├── 20240704_060236 # export
│   ├── configs.yaml
│   ├── exported_model_decoder.bin
│   ├── exported_model_decoder.xml
│   ├── exported_model_image_encoder.bin
│   └── exported_model_image_encoder.xml
├── 20240704_060350 # test
│   ├── csv
│   └── tensorboard
...
```
When a task stamp for `otx test` is `20240704_060350`, we can directly access `reference_infos` with `../.latest/train` and this relative path can be the default value I think.
If we want to use another `reference_infos` path, we can just insert different path using cli or recipe.

Reply:
Please refer api usecase
https://github.com/openvinotoolkit/training_extensions/blob/661ffee3406f1289d8412cb71f83257295225161/tests/integration/api/test_auto_configuration.py#L46-L49

Reply:
As far as I see, that option works only for default path from the recipe. The problem is that looks like a real path, although processed differently. Perhaps an empty `infer_reference_info_root` could be a marker of default path as well, but I don't mind to leave current solution as is

Reply:
In fact, I believe this case is limited to the CLI 'otx train' command. 
However, recipe yaml can be used not only in the cli, but also in the API via `Engine.from_config`. In this case, we don't see a folder called `.latest`, is it configured to be available to all? Or does it just not work in that case?

Reply:
Actually, it needs to manually update it like above comment.
https://github.com/openvinotoolkit/training_extensions/pull/3709#discussion_r1665305115